BIomes, biome generation

mezz opened the pull request:
This stuff is not easy to name because the functions are very strange. 
For review on this one, I think it's best to check this out in a branch and follow along in the code to see what things are doing, or else you might think I'm just crazy.

mezz committed:
BIomes, biome generation

copygirl commented on mappings/net/minecraft/sortme/BiomeProviderSingle.mapping:
```diff
@@ -1 +1,18 @@
+		ARG 1 z
+		ARG 2 radius
+		ARG 3 allowedBiomes
+		ARG 4 worldRand
```
Does the provider have its own `rand` field or why is it called `worldRand`?

copygirl commented on mappings/net/minecraft/world/biome/Biome.mapping:
```diff
@@ -13,13 +15,19 @@ CLASS none/ajj net/minecraft/world/biome/Biome
+		METHOD c setHeight (F)Lnone/ajj$a;
+			ARG 0 value
+		METHOD d setScale (F)Lnone/ajj$a;
+			ARG 0 scale
```
`value`
Also I'm seeing `setRainfall` doesn't use `value` either. Fix that too pls :smile:

copygirl commented on mappings/net/minecraft/world/biome/layer/BiomeLayerMapToValidBiomes.mapping:
```diff
@@ -0,0 +1,16 @@
+		ARG 0 seed
+		ARG 1 previous
+		ARG 2 levelGeneratorType
+		ARG 3 generatorOptions
```
Is this assigned to `biomeGeneratorSettings` ? Should probably either be `level` or `biomeGeneratorSettings`. Let's not call them `options` ..?

copygirl commented on mappings/net/minecraft/world/biome/layer/BiomeLayerRareVariantsAndMakeIdsValid.mapping:
```diff
@@ -0,0 +1,14 @@
+CLASS none/azo net/minecraft/world/biome/layer/BiomeLayerRareVariantsAndMakeIdsValid
```
Why `MakeIdsValid` ?

copygirl commented on mappings/none/bei.mapping:
```diff
@@ -0,0 +1,5 @@
+CLASS none/bei
+	FIELD a MAP_ICONS Lnone/kp;
```
If this is a texture it should have `_TEX` suffix.
(Though maybe we should use a different term for texture maps that contain multiple sprites/icons.)

maxpowa committed:
Misc random mappings (#120)

* Misc random mappings

* Minor changes & move WorldServerChild to sortme

* username -> userName

copygirl committed:
Rename some Gui related methods & more

lastPressedWidget
mouseDragged => onMouseDragged
handleMouseClick => onMousePressed
handleTouchEvent => onMouseReleased
handleKeyPress => onKeyPressed
handleButtonPressed => onWidgetPressed
WidgetButton.released => onReleased

mezz committed:
Add some mappings for mob pathfinding (#127)

* Add some mappings for mob pathfinding

* Fixes for review

* Fixes for review

Darkhax committed:
Several mappings (#129)

* Mapped a bunch of loot stuff.

* Mapped some random stuff.

* Mapped some debug renderers.

* Mapped a few misc things.

* element -> json

copygirl committed:
Map / rename PlayerFormatUpgrader

copygirl committed:
Fix StatManager, map StatManagerServer and such

updateStatState => set
increaseState => increase
getValue => get
EntityPlayer.getStatManager => getStats

copygirl committed:
Fix LootTable related mappings

serialze => serialize
Serializer => DeSerializer
LootTableContext:
* TARGET => Target
* source => damageSource
* setX(x) => setX(value)

shadowfacts committed:
Random block stuff

shadowfacts committed:
Random world stuff

copygirl committed:
Add some mappings for block rendering (#133)

* Add some mappings for block rendering

* Fixes some things

* Make getAoBrightness to getAmbientOcclusionBrightness

* Fix merge conflicts

* Fix a few minor things

VertexBuffer.putColorRGB => putColor
BlockRenderer hashcode parameter => variationSeed
Revert to IBlockView.getCombinedLight

copygirl committed:
Merge branch 'biomes' of git://github.com/mezz/pomf into mezz-biomes

copygirl committed:
Fix a few things
